[en] sync cluster-ns-auth (#4472)
DaoCloud/DaoCloud-docs: docs/en/docs/kpanda/user-guide/permissions/custom-kpanda-role.md

not be modified by users.
To support more flexible permission settings and to meet the customized needs for system roles,
now you can modify RBAC rules for system roles such as cluster admin, ns admin, ns editor, ns viewer.

The following example demonstrates how to add a new ns-view rule, granting the authority to delete 
workload deployments. Similar operations can be performed for other rules.

## Prerequisites

Before adding RBAC rules to system roles, the following prerequisites must be met:

- Container management version v0.27.0 and above.
- [Integrated Kubernetes cluster](../clusters/integrate-cluster.md) or
  [created Kubernetes cluster](../clusters/create-cluster.md), and able to access the cluster's UI interface.
- Completed creation of a [namespace](../namespaces/createns.md) and [user account](../../../ghippo/user-guide/access-control/user.md), 
  and the granting of [NS Viewer](./permission-brief.md#ns-viewer).
  For details, refer to [namespace authorization](./cluster-ns-auth.md).

!!! note

    - RBAC rules **only need to be added** in the Global Cluster, and the Kpanda controller will synchronize 

>       those added rules to all connected subclusters. Synchronization may take some time to complete.

Reviewer: ```suggestion
      those added rules to all integrated subclusters. Synchronization may take some time to complete.
```

    - RBAC rules **can only be added** in the Global Cluster. RBAC rules added in subclusters 
      will be overridden by the built-in role permissions of the Global Cluster.
    - Only ClusterRoles with fixed Label are supported for adding rules. Replacing or deleting rules 
      is not supported, nor is adding rules by using role. The correspondence between built-in roles and 
      ClusterRole Label created by users is as follows.

        ```output
        cluster-admin: rbac.kpanda.io/role-template-cluster-admin: "true"
        cluster-edit: rbac.kpanda.io/role-template-cluster-edit: "true"
        cluster-view: rbac.kpanda.io/role-template-cluster-view: "true"
        ns-admin: rbac.kpanda.io/role-template-ns-admin: "true"
        ns-edit: rbac.kpanda.io/role-template-ns-edit: "true"
        ns-view: rbac.kpanda.io/role-template-ns-view: "true"
        ```

## Steps

> 1. Users with admin or cluster admin permissions [create stateless workloads](../workloads/create-deployment.md)

Reviewer: ```suggestion
1. [Create a deployment](../workloads/create-deployment.md) by a user with `admin` or `cluster admin` permissions.
```

    ![image-20240514112742395](../images/create-depolyment.png)

> 1. Users grant ns-viewer and posess namespace ns-view authorization

Reviewer: ```suggestion
1. Grant a user the `ns-viewer` role to provide them with the `ns-view` permission.
```

  
    ![image-20240514113009311](../images/permisson02.png)

> 1. Switch the login user to ns-viewer, open the control panel to get the token corresponding to the ns-viewer user, 

Reviewer: ```suggestion
1. Switch the login user to ns-viewer, open the console to get the token for the ns-viewer user, 
```

>    and use curl to request to delete the above deployment nginx, finding that there is no deletion permission.

Reviewer: ```suggestion
   and use `curl` to request and delete the nginx deployment mentioned above. However,
   a prompt appears as below, indicating the user doesn't have permission to delete it.
```

    {"code":7,"message":"[RBAC] delete resources(deployments: nginx) is forbidden for user(ns-viewer) in cluster(cluster-member)","details":[]}[root@master-01 ~]#
    [root@master-01 ~]#
    ```

> 1. Create the following ClusterRole on the global cluster, as shown in the yaml below.

Reviewer: ```suggestion
1. Create a ClusterRole on the global cluster, as shown in the yaml below.
```

    ```yaml
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: append-ns-view # (1)!
      labels:
        rbac.kpanda.io/role-template-ns-view: "true" # (2)!
    rules:
      - apiGroups: [ "apps" ]
        resources: [ "deployments" ]
        verbs: [ "delete" ]
    ```

    1. This field value can be arbitrarily specified, as long as it is not duplicated and complies with 
       the Kubernetes resource naming conventions.
    2. When adding rules to different roles, make sure to apply different labels.

> 1. Wait for the kpanda controller to add the user creation rule to the built-in role ns-view. Check 

Reviewer: ```suggestion
1. Wait for the kpanda controller to add a rule of user creation to the built-in role: ns-viewer. Check 
```

>    the proper built-in role to see if the rules added in the previous step are present.

Reviewer: ```suggestion
   then you can check if the rules added in the previous step are present for ns-viewer. 
```

    ```bash
    [root@master-01 ~]# kubectl get clusterrole role-template-ns-view -oyaml|grep deployments -C 10|tail -n 6
    ```
    ```yaml
    - apiGroups:
      - apps
      resources:
      - deployments
      verbs:
      - delete
    ```

> 1. When using curl again to request the deletion of the aforementioned deployment nginx, this time the deletion 

Reviewer: ```suggestion
1. When using curl again to request the deletion of the aforementioned nginx deployment, this time the deletion 
```

>    was successful. This means that ns-viewer has successfully added the rule to delete Deployments.

Reviewer: ```suggestion
   was successful. This means that ns-viewer has successfully added the rule to delete deployments.
```

> # Increase Kpanda Built-in-role Rbac Rules

Reviewer: ```suggestion
# Add RBAC Rules in Container Management
```

*[kpanda]: A development codename for container management
   

> The permission points (rbac rules) for kpanda's built-in roles are pre-defined and cannot be modified by users. Any changes made to the built-in role permissions will be reverted to the predefined permissions by the kpanda controller. To support more flexible permission configurations and to meet the customized needs for system roles, kpanda allows the modification of permission points for built-in system roles (cluster admin, ns admin, ns edit, ns view).

Reviewer: ```suggestion
The RBAC rules for those built-in roles in container management are pre-defined and cannot be modified by users.
Previously, any changes made to the rules will be reverted.
To support more flexible permission settings and to meet the customized needs for system roles,
now you can modify allows  RBAC rules for system roles such as cluster admin, ns admin, ns editor, ns viewer.
```

> # Adding RBAC Rules in Container Management

Reviewer: ```suggestion
# Adding RBAC Rules to System Roles
```

*[kpanda]: A development codename for container management
   

> The RBAC rules for those built-in roles in container management are pre-defined and cannot be modified by users.

Reviewer: ```suggestion
In the past, the RBAC rules for those built-in roles in container management were pre-defined and could not be modified by users.
```

!!! note

    - Permission points **only need to be added** in the Global Cluster, and the Kpanda controller will synchronize those added permissions to all connected subclusters. Synchronization may take some time to complete.
    - Permission points **can only be added** in the Global Cluster. Permission points added in subclusters will be overridden by the built-in role permission points of the Global Cluster.
    - Only ClusterRoles with fixed Label are supported for adding permissions. Replacing or deleting permissions is not supported, nor is adding permissions using role. The correspondence between built-in roles and ClusterRole Label created by users is as follows.

        ```output
        cluster-admin: rbac.kpanda.io/role-template-cluster-admin: "true"
        cluster-edit: rbac.kpanda.io/role-template-cluster-edit: "true"
        cluster-view: rbac.kpanda.io/role-template-cluster-view: "true"
        ns-admin: rbac.kpanda.io/role-template-ns-admin: "true"
        ns-edit: rbac.kpanda.io/role-template-ns-edit: "true"
        ns-view: rbac.kpanda.io/role-template-ns-view: "true"
        ```

> ## Operation Steps

Reviewer: ```suggestion
## Steps
```
